src/main.cpp

```diff
@@ -2784,6 +2786,42 @@ void static UpdateTip(CBlockIndex *pindexNew) {
       Checkpoints::GuessVerificationProgress(chainParams.Checkpoints(), chainActive.Tip()), pcoinsTip->DynamicMemoryUsage() * (1.0 / (1<<20)), pcoinsTip->GetCacheSize());
 
     cvBlockChange.notify_all();
+    
+    
+
+    // MVHF-BU
+    const int MVHF_FORK_BLOCK = 50000;
+
+    //// 	Auto Backup At Block 		////
+    // Test autobackupwalletpath argument
+    if (!fAutoBackupDone)
+    {
+    	std::string strWalletBackupFile = GetArg("-autobackupwalletpath", "");
+
+		int BackupBlock = GetArg("-autobackupblock", MVHF_FORK_BLOCK - 1);
```

Comment:
MVHF-BU-SW-REQ-10-2 is implemented exactly as current documentation.

After the block becomes the best block in the UpdateTip function then the backup happens. So if the backup block is 200 then the backup triggers after 200 becomes best block and before 201.